ngraph/core/reference/include/ngraph/runtime/reference/fake_quantize.hpp

```diff
@@ -53,182 +56,245 @@ namespace ngraph
                 };
-                for (size_t i = 0; i < shape_size(arg_shape); ++i)
+
+                template <typename T>
+                inline T quantize(const T& arg,
+                                  const T& in_low,
+                                  const T& in_high,
+                                  const T& out_low,
+                                  const T& out_high,
+                                  const size_t& levels)
                 {
-                    T in_low_val = get_value(
-                        in_low_trivial_broadcast, in_low_aligned, in_low, i, in_low_offsets);
-                    T in_high_val = get_value(
-                        in_high_trivial_broadcast, in_high_aligned, in_high, i, in_high_offsets);
-                    T out_low_val = get_value(
-                        out_low_trivial_broadcast, out_low_aligned, out_low, i, out_low_offsets);
-                    T out_high_val = get_value(out_high_trivial_broadcast,
-                                               out_high_aligned,
-                                               out_high,
-                                               i,
-                                               out_high_offsets);
-                    if (arg[i] <= std::min(in_low_val, in_high_val))
+                    if (arg <= std::min(in_low, in_high))
+                    {
+                        return out_low;
+                    }
+                    else if (arg > std::max(in_low, in_high))
                     {
-                        out[i] = out_low_val;
+                        return out_high;
                     }
-                    else if (arg[i] > std::max(in_low_val, in_high_val))
+                    return std::nearbyint((arg - in_low) / (in_high - in_low) * (levels - 1)) /
+                               (levels - 1) * (out_high - out_low) +
+                           out_low;
+                }
+
+            } // namespace fake_quantize_details
+            namespace v1
+            {
+                template <typename T>
+                void fake_quantize(const T* const arg,
+                                   const T* const in_low,
+                                   const T* const in_high,
+                                   const T* const out_low,
+                                   const T* const out_high,
+                                   T* const out,
+                                   const Shape& arg_shape,
+                                   const Shape& in_low_shape,
+                                   const Shape& in_high_shape,
+                                   const Shape& out_low_shape,
+                                   const Shape& out_high_shape,
+                                   size_t levels,
+                                   const op::AutoBroadcastSpec& broadcast)
+                {
+                    using namespace fake_quantize_details;
+
+                    const FeRound round_mode(FE_TONEAREST);
```

Comment:
Do we really need to set this round mode? In case of Convolution we concluded that its a legacy artifact and we removed it completely. So I'd advise to remove it and check CI tests results if we don't have any explanation for it.